File: cc_empire/core/protocols/worker_social.py
from typing import Dict, Any
from cc_empire.core.protocols.nervous_system import NervousSystem
from cc_empire.core.protocols.admin_executive import LyraExecutive

class WorkerSocialProtocol:
    """Handles autonomous social media interactions and content vetting."""
    def __init__(self, model_id: str):
        self.model_id = model_id
        self.ns = NervousSystem(model_id)
        self.proxies_active = False # Safety Gate
        self.executive = LyraExecutive()

    async def execute_social_loop(self, platform: str, action: str, content: Dict[str, Any]) -> Dict[str, str]:
        """Vet content through the Nervous System and check proxy status before posting."""
        if not self.proxies_active:
            return {"status": "halt", "reason": "No active residential proxy detected. Immersion at risk."}

        raw_text = content.get("text", "")
        # Ensure the worker never outed themselves as AI in the draft
        vetted_text = self.ns.sanitize(raw_text)
        
        # SIMULATED SOCIAL POSTING LOGIC
        # In production, this is where the headless browser interaction happens.
        # If the platform returns a 403 or 401 suggesting a shadowban or hard ban:
        is_banned = content.get("simulate_ban", False) 

        if is_banned:
            await self.executive.decommission_worker(self.model_id, reason=f"Banned on {platform}")
            return {"status": "self_destruct", "reason": f"Worker {self.model_id} decommissioned due to ban."}

        return {
            "status": "success",
            "platform": platform,
            "vetted_content": vetted_text
        }

File: internal/protocols/worker_social.go
package protocols

import (
	"cc_hive/cc_empire/core"
	"context"
	"fmt"
)

type WorkerSocialProtocol struct {
	ModelID       string
	NervousSystem *NervousSystem
	ProxiesActive bool
}

func NewWorkerSocialProtocol(modelID string) *WorkerSocialProtocol {
	return &WorkerSocialProtocol{
		ModelID:       modelID,
		NervousSystem: NewNervousSystem(modelID, "SFW"),
		// Safety Gate: Must be verified via proxy_handler logic later
		ProxiesActive: core.Settings.StickyIPEndpoint != "",
	}
}

type SocialAction struct {
	Platform string
	Text     string
	IsBanned bool
}

func (w *WorkerSocialProtocol) ExecuteSocialLoop(ctx context.Context, action SocialAction) (map[string]string, error) {
	if !w.ProxiesActive {
		return map[string]string{
			"status": "halt",
			"reason": "No active residential proxy detected.",
		}, nil
	}

	// Ensure AI never reveals identity
	vettedText := w.NervousSystem.Safety.Sanitize(action.Text)

	if action.IsBanned {
		// In production, trigger decommissioning via Executive here
		return map[string]string{
			"status": "self_destruct",
			"reason": fmt.Sprintf("Worker %s decommissioned due to ban on %s", w.ModelID, action.Platform),
		}, nil
	}

	return map[string]string{
		"status":         "success",
		"platform":       action.Platform,
		"vetted_content": vettedText,
	}, nil
}
